_layouts/post.html

```diff
@@ -1,10 +1,17 @@
 ---
 layout: default
 ---
+
+{% include function-getContributor type="name" search=page.author %}
+
 <section id="section-page">
   <header class="main">
     <h1 class="post-title">{{ page.title }}</h1>
-    <p>{{ page.date | date_to_string }} · by <a href="{{page.author_link}}">{{page.author}}</a></p>
+    {% if contributor.homepage %}
+      <p>{{ page.date | date_to_string }} · by <a href="{{ contributor.homepage }}">{{page.author}}</a></p>
```

Comment:
I _think_ we could do this without an `if` but I'm not sure that's necessarily more readable, what do you think?

```html
<p>
{% assign homepage = contributor.homepage | default: contributor.github %}
{{ page.date | date_to_string }} · by <a href="{{ homepage }}">{{page.author}}</a>
</p>
```

Reply:
i'd call the variable `author_url`

but I prefer your version!

Reply:
It seems cleaner, only thing is the way you do it, there will be a mistake in github url. You must build the github url in default (btw is there string interpollation in jekyll?).

Reply:
i've committed the cleaner version 👍 thanks for the input all :)

Reply:
> btw is there string interpollation in jekyll?).

@gemantzu [Liquid](https://shopify.github.io/liquid/) (the templating engine of Jekyll) has [append](https://shopify.github.io/liquid/filters/append/) and [prepend](https://shopify.github.io/liquid/filters/prepend/) functions